### ISSUE TITLE
Cannot read property 'trim' of undefined

### ISSUE THREAD

**Issue by auctiontrac** [huserben/TfsExtensions#65]
When running a triggered build I get the following error

![image](https://user-images.githubusercontent.com/7092400/38820255-62652536-416b-11e8-8ba9-d538cc16a381.png)

I think it may have something to do with our builds having spaces in their names.

**Comment by huserben** [comment]
Hi

could you please post the full log of the Task.
Ideally you set the system.debug variable to true before, so I can see in detail whats happening.

However the spaces should not be an issue.

**Comment by thomasonb** [comment]
Hi sorry when I posted that original issue I was in the wrong account but here are is the log for that task.

[TriggeredBuildLog.txt](https://github.com/huserben/TfsExtensions/files/1916270/TriggeredBuildLog.txt)

**Comment by thomasonb** [comment]
Another thing that is interesting (I can make a new issue is necessary) is that even though I've told it to fail the build when the tasks fail, it still continues with the build in this instance. This could be a side effect of the error I receive since it's failing before the tasks are fired?

**Comment by huserben** [comment]
Hi

thanks for the logs. It shows that at least the build seems to be triggered - I'll need to check in more detail what exactly is wrong and try to recreate the problem locally. The ony "special" thing I can see are the "-" in the name of the build definition, I will see whether this is problematic.

What exactly do you mean with "told it to fail when the task fails"?

**Comment by thomasonb** [comment]
Ok so I modified what you asked and that did fix the issue, thank you.

As for the other question, under AdvancedConfiguration section there is the following checkbox.

![image](https://user-images.githubusercontent.com/3091040/38826043-a08a8502-417c-11e8-9731-dcfa93b83ac5.png)

When I was getting the 'trim' error the build would continue, I would expect the build to fail right then. I have also already checked the phase after this task and they are set to only continue if all previous phases have succeded

![image](https://user-images.githubusercontent.com/3091040/38826155-fe46df9c-417c-11e8-84a0-198d6136b0e7.png)

This is what the process looks like I would expect the "Phase 1" to just fail immediately, but maybe it doesn't work how I thought it does.

![image](https://user-images.githubusercontent.com/3091040/38826254-3bbad63a-417d-11e8-8591-c9d3976cef39.png)

**Comment by huserben** [comment]
This option is failing the Task in case the build you triggered has failed. So this has nothing todo with how to proceed after the Task has failed (refer as well to the documentation "Wait for Completion of Triggered Builds" in the [Task Overview](https://marketplace.visualstudio.com/items?itemName=benjhuser.tfs-extensions-build-tasks).

What happens is that the Task is failing "properly" (it is in the failed state as the red "X" indication marks it in your screenshot).
How/Why it's then proceeding I cannot tell, but from the Task perspective it seems to be ok, as it is failed. The rest is TFS/VSTS configuration and out of my hands, as I don't have any experience with the Phases. But looking at the configuration I would as well expect that it fails immediately.

The only thing I could see is if the Phases do run in parallel. I think you can specifiy parallelism and as well make the phases depend on each other if needed. Maybe that's something to look at.

**Comment by thomasonb** [comment]
Ok I will look into that, thanks for your help!

**Comment by huserben** [comment]
No problem. I just now updated the task so it will print out a more meaningful message in case wrongly specified parameters are setup.

Thank you for your input and please don't hestitate to open another issue if you find a problem, you have a change request or simply a question.
